base/range.jl

```diff
@@ -860,13 +860,19 @@ end
 
 median(r::Range{<:Real}) = mean(r)
 
-function in(x, r::Range)
+function in(x::Real, r::Range{<:Real})
+    n = step(r) == 0 ? 1 : round(Integer,(x-first(r))/step(r))+1
+    n >= 1 && n <= length(r) && r[n] == x
+end
+# This method needs to be defined separately since only -(::T, ::T) needs to be supported
+# to create a range, but not necessarily -(::T, ::Real)
+function in(x::T, r::Range{<:T}) where {T}
```

Comment:
I've changed `<:T` to `T`  and it doesn't seem to generate ambiguities now, so let's go with that.